Source/Core/ScreenSpaceEventHandler.js

```diff
@@ -934,12 +934,13 @@ function ScreenSpaceEventHandler(element) {
   registerListeners(this);
 }
 
+// TODO: document callback signature for each type of event
```

Comment:
Do you plan on tackling this TODO?

Reply:
It's a little complicated.  I'm making typedefs and a callback to describe the event types that can be passed to a callback, but the right way to solve it involves a number of overloads for `setInputAction`, at least one per type of callback (of which there are five).  JSDoc overloads don't work very well (see previous PR with `exportKML`), and cause issues with `prettier`.

For now, I'm going to push out a version with a type-union callback parameter.  I'll look at doing overloads but will probably hold off unless/until there's a better way to implement them.

Reply:
OK I got this 100% working, I'm pretty sure, but it is going to take a little explaining.  Overloads in JSDoc are a bit hairy; there are pros and cons to my approach.

On the pro side, Typescript users can now write `viewer.screenSpaceEventHandler.setInputAction(evt => doStuff(Cartographic.fromCartesian(evt.position)), ScreenSpaceEventType.MOUSE_DOWN)` and the `position` property will be correctly typed.  This is pretty great!  It's the same way that TS lets you write e.g. `input.addEventListener("input", evt => doStuff(evt.inputType))` and get strong typing where `evt` is automatically an `InputEvent`.

On the con side, there's a bunch of extra typedefs and `@callback`s that shouldn't really need to be named.  It takes up extra space in the source file (though it's all comments so it should minify out for production) and adds some entries to the docs of questionable utility.

I did update the "container" template in the docs so that any function declaration with a `@variation` tag greater than 1 is not printed in the docs.  This avoids a problem described in the previous version of this PR where the overridden function would get one entire doc entry for each override -- `@variation 1` is the one that gets printed.  (I just found `Documentation\Contributors\DocumentationGuide`, will add a quick description of this behavior there!)

Reply:
I have to say I'm a bit skeptical about this approach. @kring, would you be able to weigh in here?

Reply:
If we need to back this out, let me know.  I really dislike the syntax we're forced to use to implement overloading via (tsd-)JSDoc; in TypeScript it would be one line per overload instead of 10+, and it's *much* easier to read.

That said, the resulting overloaded types really are quite helpful, especially when registering event handlers via a one-liner lambda function.  The alternative is that callbacks always type as a plain `Function` (`(...args: any[]) => void`) which, when the TS compiler is in "strict" mode, means that every callback has to explicitly type the arguments.  Worse, as far as I can tell, they aren't actually written out explicitly anywhere in (the current version of) the docs.

Reply:
Thanks @thw0rted, let's back it out so we can focus on getting the other improvements in this PR merged.

Feel free to open an issue or a separate PR for the overloaded types, and we can address them separately. :+1:

Reply:
Please take a look and let me know if it's what you had in mind.  (I also caught one more little error, I just keep this branch open and correct stuff when it bites me.)